bleach/html5lib_shim.py

```diff
@@ -256,7 +256,7 @@ def __iter__(self):
                     yield token
 
                 elif ((last_error_token['data'] == 'expected-closing-tag-but-got-char' and
-                     token['data'].lower().strip() not in self.parser.tags)):
+                     (self.parser.tags is None or token['data'].lower().strip() not in self.parser.tags))):
```

Comment:
I think this isn't the right fix. If the user explicitly creates a parser that has `tags=None`, then I'm pretty sure it should not look at tags at all and do whatever html5lib does by default.

I think we need to carry that through here, too. So then this should be:

```
self.parser.tags is not None and token['data'].lower().strip() not in self.parser.tags
```

That results in the expected output of the added test being different.

Reply:
I've updated fix and test as you suggested